integration/mediation-tests/tests-common/integration-test-utils/src/main/java/org/wso2/esb/integration/common/utils/servers/GreenMailServer.java

```diff
@@ -0,0 +1,239 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.esb.integration.common.utils.servers;
+
+import com.icegreen.greenmail.user.GreenMailUser;
+import com.icegreen.greenmail.util.GreenMail;
+import com.icegreen.greenmail.util.ServerSetup;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.esb.integration.common.utils.exception.ESBMailTransportIntegrationTestException;
+
+import java.util.Properties;
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.mail.URLName;
+import javax.mail.search.AndTerm;
+import javax.mail.search.BodyTerm;
+import javax.mail.search.SearchTerm;
+import javax.mail.search.SubjectTerm;
+
+/**
+ * Util class for GreenMail Server
+ */
+public class GreenMailServer {
+    protected static Log log = LogFactory.getLog(GreenMailServer.class);
+    private static GreenMail greenMail;
+    private static GreenMailUser greenMailUser;
+    private static String EMAIL_INBOX = "INBOX";
+    private static String EMAIL_TRASH = "Trash";
+    private static int WAIT_TIME_MS = 180 * 1000;
+
+    /**
+     * Start the server and add the user
+     */
+    public void startServer() {
+        ServerSetup popSetup = new ServerSetup(3110, "localhost", "pop3");
+        ServerSetup imapSetup = new ServerSetup(3143, "localhost", "imap");
+        ServerSetup[] setups = new ServerSetup[2];
+        setups[0] = popSetup;
+        setups[1] = imapSetup;
+        greenMail = new GreenMail(setups);
+        greenMail.start();
+        greenMailUser = greenMail.setUser("wso2@localhost", "wso2", "wso2");
```

Comment:
let's define constants to the repetitively used strings. We can have a separate constant class if the same is being used by multiple classes

Reply:
done